br/pkg/restore/internal/prealloc_db/db.go

```diff
@@ -280,26 +280,29 @@ func (db *DB) CreateTablePostRestore(ctx context.Context, table *metautil.Table,
 	return nil
 }
 
-func (db *DB) tableIDAllocFilter() ddl.AllocTableIDIf {
-	return func(ti *model.TableInfo) bool {
-		if db.preallocedIDs == nil {
-			return true
-		}
-		prealloced := db.preallocedIDs.PreallocedFor(ti)
-		if prealloced {
-			log.Info("reusing table ID", zap.Stringer("table", ti.Name))
-		}
-		return !prealloced
+func (db *DB) canReuseTableID(ti *model.TableInfo) bool {
+	if db.preallocedIDs == nil {
+		return false
+	}
+	prealloced := db.preallocedIDs.PreallocedFor(ti)
+	if prealloced {
+		log.Info("reusing table ID", zap.Stringer("table", ti.Name))
 	}
+	return prealloced
 }
 
 // CreateTables execute a internal CREATE TABLES.
 func (db *DB) CreateTables(ctx context.Context, tables []*metautil.Table,
 	ddlTables map[restore.UniqueTableName]bool, supportPolicy bool, policyMap *sync.Map) error {
 	if batchSession, ok := db.se.(glue.BatchCreateTableSession); ok {
-		m := map[string][]*model.TableInfo{}
+		idReusableTbls := map[string][]*model.TableInfo{}
+		idNonReusableTbls := map[string][]*model.TableInfo{}
 		for _, table := range tables {
-			m[table.DB.Name.L] = append(m[table.DB.Name.L], table.Info)
+			if db.canReuseTableID(table.Info) {
```

Comment:
It splits the batched `tables` into `idResuableTbls` and `idNonResuableTbls`. I prefer to splits the `unbatched tables` before here
https://github.com/pingcap/tidb/blob/2088b342e8f04f362d84cd54ab99f9fc7dc56ddf/br/pkg/restore/snap_client/pipeline_items.go#L416